server/src/main/java/io/crate/analyze/relations/AliasedAnalyzedRelation.java

```diff
@@ -165,7 +167,7 @@ public Symbol resolveField(ScopedSymbol field) {
         }
         assert childColumnName != null
             : "If a ScopedSymbol has been retrieved via `getField`, it must be possible to get the columnIdent";
-        var result = relation.getField(childColumnName, Operation.READ);
+        var result = relation.getField(childColumnName, Operation.READ, true);
```

Comment:
Shouldn't the `true` here use the value from the session setting?

Reply:
I am thinking it is unnecessary because this logic is part of `resolveField(ScopedSymbol field)` where `ScopedSymbol` represents that it has been analyzed and thus no need further error checking. I can overload it without the setting that it is more readable.